lib/datadog/core/environment/object_space.rb

```diff
@@ -0,0 +1,21 @@
+require 'objspace'
+
+module Datadog
+  module Core
+    module Environment
+      # Retrieves garbage collection statistics
+      module ObjectSpace
+        module_function
+
+        def estimate_bytesize(object)
+          return 0 unless ::ObjectSpace.respond_to?(:memsize_of)
+
+          # Rough calculation of bytesize; not very accurate.
+          object.instance_variables.inject(::ObjectSpace.memsize_of(object)) do |sum, var|
+            sum + ::ObjectSpace.memsize_of(object.instance_variable_get(var))
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
This only seems to be used in specs. I suggest we take the opportunity to move it out of `lib/` and into `spec/`.

Reply:
This one is not blocking, but it's a really nice clean up :)